# HG changeset patch
# User omajid
# Date 1401638044 14400
#      Sun Jun 01 11:54:04 2014 -0400
# Node ID fd78f7e83f3f1ce7d363b681b97583138218afcb
# Parent  8d0af21b5b9789f97b05edacc221a63f851d13b8
8044235: src.zip should include all sources
Reviewed-by: andrew, dholmes

diff --git a/make/CreateJars.gmk b/make/CreateJars.gmk
--- openjdk/jdk/make/CreateJars.gmk
+++ openjdk/jdk/make/CreateJars.gmk
@@ -584,40 +584,53 @@
 
 ##########################################################################################
 
-SRC_ZIP_INCLUDES = \
-    com/sun/corba \
-    com/sun/image/codec/jpeg \
-    com/sun/imageio \
-    com/sun/java_cup \
-    com/sun/javadoc \
-    com/sun/java/swing \
-    com/sun/jmx \
-    com/sun/naming \
-    com/sun/org/apache \
-    com/sun/security/auth \
-    com/sun/security/jgss \
-    com/sun/source \
-    java \
-    javax/accessibility \
-    javax/annotation \
-    javax/imageio \
-    javax/lang \
-    javax/management \
-    javax/naming \
-    javax/print \
-    javax/rmi \
-    javax/script \
-    javax/security \
-    javax/sound \
-    javax/sql \
-    javax/swing \
-    javax/tools \
-    javax/xml \
-    org/ietf \
-    org/omg \
-    org/w3c/dom \
-    org/xml/sax \
-    #
+ifdef OPENJDK
+    SRC_ZIP_INCLUDES = \
+        com \
+        java \
+        javax \
+        jdk \
+        org \
+        sun \
+        #
+    SRC_ZIP_EXCLUDES =
+else
+    SRC_ZIP_INCLUDES = \
+        com/sun/corba \
+        com/sun/image/codec/jpeg \
+        com/sun/imageio \
+        com/sun/java_cup \
+        com/sun/javadoc \
+        com/sun/java/swing \
+        com/sun/jmx \
+        com/sun/naming \
+        com/sun/org/apache \
+        com/sun/security/auth \
+        com/sun/security/jgss \
+        com/sun/source \
+        java \
+        javax/accessibility \
+        javax/annotation \
+        javax/imageio \
+        javax/lang \
+        javax/management \
+        javax/naming \
+        javax/print \
+        javax/rmi \
+        javax/script \
+        javax/security \
+        javax/sound \
+        javax/sql \
+        javax/swing \
+        javax/tools \
+        javax/xml \
+        org/ietf \
+        org/omg \
+        org/w3c/dom \
+        org/xml/sax \
+        #
+    SRC_ZIP_EXCLUDES = javax/swing/beaninfo
+endif
 
 SRC_ZIP_SRCS = $(JDK_TOPDIR)/src/share/classes $(JDK_TOPDIR)/src/$(OPENJDK_TARGET_OS_API_DIR)/classes
 SRC_ZIP_SRCS += $(JDK_OUTPUTDIR)/gensrc
@@ -648,7 +661,7 @@
 $(eval $(call SetupZipArchive,BUILD_SRC_ZIP, \
     SRC := $(SRC_ZIP_SRCS) $(IMAGES_OUTPUTDIR)/src, \
     INCLUDES := $(SRC_ZIP_INCLUDES) launcher, \
-    EXCLUDES := javax/swing/beaninfo, \
+    EXCLUDES := $(SRC_ZIP_EXCLUDES), \
     SUFFIXES := .java .c .h, \
     ZIP := $(IMAGES_OUTPUTDIR)/src.zip, \
     EXTRA_DEPS := $(LAUNCHER_ZIP_SRC)))
